### ISSUE TITLE
How to use table name different from the default?

### ISSUE THREAD

**Issue by rainyday** [diesel-rs/diesel#2475]
How can I manually specify a table name for a Queryable item? I've looked around the documentation and github and it looks like this feature was added a few years ago in https://github.com/diesel-rs/diesel/pull/1233? Unfortunately, I'm unable to find any documentation on how to actually use this feature.
## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [x] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)

**Comment by weiznich** [comment]
`Queryable` is not related to an table in any way. `Queryable` is a trait that maps the results of an query to a user provided type, where a query does not necessarily contain exactly one table. Therefore there is no way to "use a different table name" as this does not even make sense for what `Queryable` stand for.

Closed as not an issue.

**Comment by rainyday** [comment]
@weiznich Ok, let me rephrase using the getting started example:

Given these contents of models.rs:

```rust
#[derive(Queryable)]
pub struct Post {
    pub id: i32,
    pub title: String,
    pub body: String,
    pub published: bool,
}
```

Diesel generates this schema.rs

```rust
table! {
    posts (id) {
        id -> Nullable<Integer>,
        title -> Text,
        body -> Text,
        published -> Bool,
    }
}
```

My question is: how is the table name `posts` determined and how can I configure it to be something else instead?
